apps/dashboard/app/apps/user_configuration.rb

```diff
@@ -53,4 +86,17 @@ def fetch(key_value, default_value = nil)
     # This is to mimic the Hash.fetch behaviour that only uses the default_value when key is not present
     profile_config.key?(key) ? profile_config[key] : @config.fetch(key, default_value)
   end
+
+  def add_property_methods
+    UserConfiguration::USER_PROPERTIES.each do |property|
+      define_singleton_method(property.name) do
+        environment_value = property.map_string(ENV[property.environment_name]) if property.read_from_environment?
+        environment_value.nil? ? fetch(property.name, property.default_value) : environment_value
+      end
+    end.each do |property|
+      define_singleton_method("#{property.name}?".to_sym) do
+        !send(property.name).nil?
```

Comment:
I wonder how this will play with things that you need to set to `false`. I.e, `OOD_MY_NEW_PROPERTY=false` and `my_new_property?` then returning `true`. 

A quick glance at the test cases seems to suggest this would be the behavior.

Reply:
In this implementation, the `my_new_property?` method always represent "Does the property have a value?" regardless of the type. Not sure it will be best to remove it as it might be confusing for booleans as you said.

Reply:
Is anything outside of tests using this right now?  I feel like `?` APIs generally ask the question 'is it true' not 'is it set'.

I'd ask to remove it altogether if nobody uses it yet. Maybe a pattern will form if/when there is a use case?

It's just one of those things that you'd have to remember, which would lead to confusion & bugs I'm sure.

Reply:
Makes sense. It is not used yet, I will remove.